### DISCUSSION TITLE
I dont know how to get the streamkey!

### DISCUSSION THREAD

**Discussion by danielavdalian** [zorchenhimer/MovieNight#166] [Q&A]
Hi! i'm trying to run movienight on my Truenas server, but i dont know where to find the Stream Key! 

Anyone can help me? 

Ty!

**Comment by zorchenhimer** [comment]
The stream key is not autogenerated, it is something you need to configure either in the settings file (`StreamKey` setting) or on the command line (`-k` option).

**Reply by danielavdalian** [reply]
Hi! Thank you for the reply, Okey, but now I have a problem... I dont know how to do it on my Truenas Jail Container (installed from plugin). 
I searched on my jail Shell but I can't find it...

And the StreamKey string, it can be whatever I want? like a Random String ?
Ty and sorry for my english 😵‍💫

**Reply by CptPie** [reply]
I have no experience with Truenas Jail containers but if they work anywhere close to how docker functions you should be able to (somehow) inject a configuration file into the container, here you would be able to set your stream key.

Regarding your other question: The streamkey can be whatever string you like, although i would recommend to choose a not too simple streamkey if you want to avoid other people possibly highjacking the stream.

**Reply by zorglube** [reply]
Hello ! 

I'm the one who make the ThrueNAS plugin part of MovieNight. 

> Hi! Thank you for the reply, Okey, but now I have a problem... I dont know how to do it on my Truenas Jail Container (installed from plugin). 

I'm not sure you really need to make you own Jail, but anyway you can find my first try to build a script that make a Jail with MovieNight inside. Feel free to purpose PR https://github.com/zorglube/freenas-iocage-movienight 

> I searched on my jail Shell but I can't find it...

Normally with an `# iocage list` you should see the Jail (plugin or own build) containing MovieNight one the list. 

> And the StreamKey string, it can be whatever I want? like a Random String ?

As far as I know yes. 

> Ty and sorry for my english 😵‍💫 

You english seems good.

**Reply by zorglube** [reply]
> I have no experience with Truenas Jail containers but if they work anywhere close to how docker functions you should be able to (somehow) inject a configuration file into the container, here you would be able to set your stream key.

Jail is more like Chroot running environnement, and Docker is more like Thread. 
That implie in a Jail you don't inject the configuration at runtime, you're still able to edit it trgought filesystem or by running a console into the running Jail.

**Reply by zorglube** [reply]
@danielavdalian if you whant to look how the MovieNight Plugin is configured and deployed it's here https://github.com/zorglube/iocage-plugin-movienight

**Reply by CenterfireXid** [reply]
Hi zorglube! Awesome stuff working on the plugin! I was able to get it installed on my trueNAS Core server, but I have no clue how to configure it, I'm sorta a noob to all this. Do I find a program that streams to stmp to play the video? Can I play the video through the truenas server? Do I get a link to give to my friends so we can watch the movie together and chat? is there a GUI I can edit all these settings from or will I need to do it through the TrueNAS terminal? Also is there some sort of youtube video tutorial on all this so I can learn from? I looked everywhere and am still lost

**Reply by zorglube** [reply]
> Hi zorglube! Awesome stuff working on the plugin! 

Thank ! 

> I was able to get it installed on my trueNAS Core server, but I have no clue how to configure it, I'm sorta a noob to all this. 

About the **TrueNAS Core** plugin, the documentation are [here](https://github.com/zorglube/iocage-plugin-movienight) and [here](https://github.com/zorchenhimer/MovieNight)

> Do I find a program that streams to stmp to play the video? 

You can use [OBS Project](https://obsproject.com)

> Can I play the video through the truenas server? 

You mean emit the origin video-stream from your nas ? 

> Do I get a link to give to my friends so we can watch the movie together and chat? 

That depend on you and how you TrueNAS instance is exposed on the internet. 

> is there a GUI I can edit all these settings from or will I need to do it through the TrueNAS terminal? 

Terminal. (`# iocage console [MOVIENIGHT JAIL NAME]`)

> Also is there some sort of youtube video tutorial on all this so I can learn from? 

As far as i know: **no**

> I looked everywhere and am still lost

Continue reading ans asking.
